src/Adapter/MSTest.CoreAdapter/Execution/TcmTestPropertiesProvider.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.VisualStudio.TestPlatform.MSTest.TestAdapter.Execution
+{
+    using System.Collections.Generic;
+    using TestPlatformObjectModel = Microsoft.VisualStudio.TestPlatform.ObjectModel;
+
+    /// <summary>
+    /// Reads and parses the TcmTestProperties in order to populate them in TestRunParameters.
+    /// </summary>
+    internal static class TcmTestPropertiesProvider
+    {
+        /// <summary>
+        /// Gets tcm properties from test case.
+        /// </summary>
+        /// <param name="testCase">Test case.</param>
+        /// <returns>Tcm properties.</returns>
+        public static IDictionary<TestPlatformObjectModel.TestProperty, object> GetTcmProperties(TestPlatformObjectModel.TestCase testCase)
+        {
+            var tcmProperties = new Dictionary<TestPlatformObjectModel.TestProperty, object>();
+
+            // Return empty proeprties when testCase is null or when test case id is zero.
+            if (testCase == null ||
+                testCase.GetPropertyValue<int>(Constants.TestCaseIdProperty, default(int)) == 0)
+            {
+                return tcmProperties;
```

Comment:
This is done so as to not having null check outside this method context. In case we return null, we can't iterate over it via foreach loop untill there is a null check.